neo/io/neomatlabio.py

```diff
@@ -199,131 +243,146 @@ def __init__(self, filename=None):
-        if 'block' not in d:
-            self.logger.exception('No block in ' + self.filename)
+        assert not lazy, "Does not support lazy"
+
+        d = scipy.io.loadmat(self.filename, struct_as_record=False, squeeze_me=True, mat_dtype=True)
+        if "block" not in d:
+            self.logger.exception("No block in " + self.filename)
             return None
 
-        bl_struct = d['block']
-        bl = self.create_ob_from_struct(
-            bl_struct, 'Block')
+        bl_struct = d["block"]
+        bl = self.create_ob_from_struct(bl_struct, "Block")
+        self._resolve_references(bl)
         bl.check_relationships()
         return bl
 
     def write_block(self, bl, **kargs):
         """
         Arguments:
-            bl: the block to b saved
+            bl: the block to be saved
         """
         import scipy.io
+
         bl_struct = self.create_struct_from_obj(bl)
 
         for seg in bl.segments:
             seg_struct = self.create_struct_from_obj(seg)
-            bl_struct['segments'].append(seg_struct)
-
-            for anasig in seg.analogsignals:
-                anasig_struct = self.create_struct_from_obj(anasig)
-                seg_struct['analogsignals'].append(anasig_struct)
-
-            for irrsig in seg.irregularlysampledsignals:
-                irrsig_struct = self.create_struct_from_obj(irrsig)
-                seg_struct['irregularlysampledsignals'].append(irrsig_struct)
-
-            for ea in seg.events:
-                ea_struct = self.create_struct_from_obj(ea)
-                seg_struct['events'].append(ea_struct)
-
-            for ea in seg.epochs:
-                ea_struct = self.create_struct_from_obj(ea)
-                seg_struct['epochs'].append(ea_struct)
+            bl_struct["segments"].append(seg_struct)
+
+            for container_name in seg._child_containers:
+                for child_obj in getattr(seg, container_name):
+                    child_struct = self.create_struct_from_obj(child_obj)
+                    seg_struct[container_name].append(child_struct)
+
+        for group in bl.groups:
+            group_structure = self.create_struct_from_obj(group)
+            bl_struct["groups"].append(group_structure)
+
+            for container_name in group._child_containers:
+                for child_obj in getattr(group, container_name):
+                    if isinstance(child_obj, (ChannelView, RegionOfInterest)):
+                        child_struct = self.create_struct_from_view(child_obj)
+                        group_structure[container_name].append(child_struct)
+                    else:
+                        group_structure[container_name].append(id(child_obj))
 
-            for sptr in seg.spiketrains:
-                sptr_struct = self.create_struct_from_obj(sptr)
-                seg_struct['spiketrains'].append(sptr_struct)
+        if kargs.get("debug", False):
+            breakpoint()
```

Comment:
I haven't seen this in other IOs. Is this just for you debugging right now or did you want to keep this?

Reply:
I left this in by accident, removed now.